src/CachingStream.php

```diff
@@ -60,9 +60,13 @@ public function seek($offset, $whence = SEEK_SET)
         $diff = $byte - $this->stream->getSize();
 
         if ($diff > 0) {
-            // If the seek byte is greater the number of read bytes, then read
-            // the difference of bytes to cache the bytes and inherently seek.
-            $this->read($diff);
+            // Read the remoteStream until we have read in at least the amount of
+            // bytes requested, or we reach the end of the file.
+            while ($diff > 0 && !($this->stream->eof(
```

Comment:
Do you need to call `$this->stream->eof()` here? It seems like you should only stop reading if the remote stream has been fully read.